test/sol5/kyberNetworkProxy.js

```diff
@@ -578,6 +578,36 @@ contract('KyberNetworkProxy', function(accounts) {
             });
         }
     });
+
+    describe("test swap function (backward compatible)", async () => {
+        it("swapEtherToToken", async () => {
+            let initialTakerBalance = await tokens[1].balanceOf(taker);
+            let tokenAdd = tokens[1].address;
+            let srcQty = (new BN(2)).mul((new BN(10)).pow(ethDecimals));
+            let proxyRate = await networkProxy.getExpectedRate(ethAddress, tokenAdd, srcQty);
+            await networkProxy.swapEtherToToken(tokenAdd, proxyRate.worstRate, { from: taker, value: srcQty });
+            let dstQty = Helper.calcDstQty(srcQty, ethDecimals, tokenDecimals[1], proxyRate.expectedRate);
+            let expectedBalance = new BN(initialTakerBalance).add(dstQty);
+            await Helper.assertSameTokenBalance(taker, tokens[1], expectedBalance);
+        });
+        it("swapTokenToEther", async () => {
+            let initialTakerBalance = await Helper.getBalancePromise(taker);
+            let srcToken = tokens[2]
+            let tokenAdd = tokens[2].address;
+            let srcQty = (new BN(5)).mul((new BN(10)).pow(tokenDecimals[2]));
+            await srcToken.approve(networkProxy.address, srcQty, { from: taker })
+            let proxyRate = await networkProxy.getExpectedRate(tokenAdd, ethAddress, srcQty);
+            let txGasPrice = new BN(2).mul(new BN(10).pow(new BN(10)));
```

Comment:
why setting gas price value?

better test 1 thing each test.
if u want to test it (and it we actually should)
do it please in another test

Reply:
I try to calculate expectedBalance := initialBalance + dstQty - txGasPrice * tx gas used.
However, the taker balance is different from expected one
Currently, I remove this checked. But if I check with this block of code, it will fail. 
Pls give me some suggestions why.
```
        it("swapTokenToEther", async () => {
            let initialTakerBalance = await Helper.getBalancePromise(taker);
            let srcToken = tokens[1]
            let tokenAdd = tokens[1].address;
            let srcQty = (new BN(5)).mul((new BN(10)).pow(tokenDecimals[1]));
            let txGasPrice = new BN(2).mul(new BN(10).pow(new BN(10)));
            await srcToken.approve(networkProxy.address, srcQty, { from: taker , gasPrice: txGasPrice});
            let proxyRate = await networkProxy.getExpectedRate(tokenAdd, ethAddress, srcQty);
            Helper.assertEqual(proxyRate.worstRate, proxyRate.expectedRate.mul(new BN(97)).div(new BN(100)));
            let txResult = await networkProxy.swapTokenToEther(tokenAdd, srcQty, proxyRate.expectedRate, { from: taker });
            let dstQty = Helper.calcDstQty(srcQty, tokenDecimals[1], ethDecimals, proxyRate.expectedRate);
            let expectedBalance = new BN(initialTakerBalance).add(dstQty).sub(txGasPrice.mul(new BN(txResult.receipt.gasUsed)));
            await Helper.assertSameEtherBalance(taker, expectedBalance);
        });
```

Reply:
Sorry this is correct version
```
        it("swapTokenToEther", async () => {
            let srcToken = tokens[1]
            let tokenAdd = tokens[1].address;
            let srcQty = (new BN(5)).mul((new BN(10)).pow(tokenDecimals[1]));
            let txGasPrice = new BN(2).mul(new BN(5).pow(new BN(10)));
            await srcToken.transfer(taker, srcQty);
            await srcToken.approve(networkProxy.address, srcQty, { from: taker });
            let initialTakerBalance = await Helper.getBalancePromise(taker);
            let proxyRate = await networkProxy.getExpectedRate(tokenAdd, ethAddress, srcQty);
            Helper.assertEqual(proxyRate.worstRate, proxyRate.expectedRate.mul(new BN(97)).div(new BN(100)));
            let txResult = await networkProxy.swapTokenToEther(tokenAdd, srcQty, proxyRate.expectedRate, { from: taker, gasPrice: txGasPrice });
            let dstQty = Helper.calcDstQty(srcQty, tokenDecimals[1], ethDecimals, proxyRate.expectedRate);
            let expectedBalance = new BN(initialTakerBalance).add(dstQty).sub(new BN(txGasPrice).mul(new BN(txResult.receipt.gasUsed)));
            await Helper.assertSameEtherBalance(taker, expectedBalance);
        });
```
the different is 2 wei